release-tools/stochss_ami_manager.py

```diff
@@ -0,0 +1,548 @@
+        while (True):
+            time.sleep(5)
+            tmp_log_stdout_file = tempfile.TemporaryFile()
+            tmp_log_stderr_file = tempfile.TemporaryFile()
+
+            remote_cmd = get_remote_command(user=self.instance_user, ip=self.instance_ip, key_file=self.key_file,
+                                            command="echo Instance {0} with ip {1} is up!".format(self.instance_id,
+                                                                                                  self.instance_ip))
+            shell_cmd = ShellCommand(remote_cmd, stdout=tmp_log_stdout_file, stderr=tmp_log_stderr_file)
+            shell_cmd.run(timeout=5, silent=True)
+
+            if self.verbose:
+                tmp_log_stderr_file.seek(0)
+                print >> 2, tmp_log_stderr_file.read()
+
+            tmp_log_stderr_file.close()
+
+            tmp_log_stdout_file.seek(0)
+            output = tmp_log_stdout_file.read().strip()
+            tmp_log_stdout_file.close()
+
+            if output == "Instance {0} with ip {1} is up!".format(self.instance_id, self.instance_ip):
+                print output
+                break
+
+    def __reboot_instance(self):
+        print '=================================================='
+        print 'Rebooting Instance {0}...'.format(self.instance_id)
+        self.ec2_connection.reboot_instances(instance_ids=[self.instance_id])
+        self.__wait_until_successful_ssh()
+
+
+    def __install_python_packages(self):
+        header = 'Installing Python Packages...'
+        print '=================================================='
+        print header
+
+        commands = []
+        for package in self.python_packages:
+            if package.has_key('version'):
+                commands.append('sudo pip uninstall -y {0}'.format(package['name']))
+                commands.append('sudo pip install {0}=={1}'.format(package['name'], package['version']))
+            else:
+                commands.append('sudo pip install {0}'.format(package['name']))
+
+        command = ';'.join(commands)
+        self.__run_remote_command(command=command, log_header=header)
+
+
+    def __download_stochss_repo(self):
+        header = 'Downloading StochSS...'
+        print '=================================================='
+        print header
+        commands = ['rm -rf stochss',
+                    'git clone --recursive {0}'.format(self.git_repo['url'])]
+
+        if self.git_repo.has_key('branch'):
+            commands.append('cd stochss')
+            commands.append('git checkout {0}'.format(self.git_repo['branch']))
+
+        command = ';'.join(commands)
+        self.__run_remote_command(command=command, log_header=header)
+
+    def __compile_stochss(self):
+        header = 'Compiling StochSS...'
+        print '=================================================='
+        print header
+        commands = ['cd stochss',
+                    "sed -i '\|launchapp|d' run.ubuntu.sh",
```

Comment:
We discussed splitting "run.ubuntu.sh" into 'run' and 'install', so that this line is not necessary and the whole system will be more maintainable.